pkgs/tools/networking/picosnitch/default.nix

```diff
@@ -0,0 +1,61 @@
+{ lib
+, python3
+, bcc
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "picosnitch";
+  version = "0.12.0";
+
+  src = python3.pkgs.fetchPypi {
+    inherit pname version;
+    sha256 = "b87654b4b92e28cf5418388ba1d3165b9fa9b17ba91af2a1a942f059128f68bc";
+  };
+
+  propagatedBuildInputs = with python3.pkgs; [
+    setuptools
+    bcc
+    psutil
+    dbus-python
+    requests
+    pandas
+    plotly
+    dash
+  ];
+
+  serviceFile = ''
```

Comment:
Normally we create all service files via nixos modules

Reply:
Sorry I'm not sure how to make this service file into a nixos module, I did change fetchPypi to fetchurl so that it can grab the service file directly from the source repo though which is how I see most packages do it, is that alright or am I still missing something?

Reply:
Thats also fine.

I would recommend to look at a nixos service which does not have many options or is very long https://github.com/NixOS/nixpkgs/blob/bef33b1411413f7ae3cc3b93239d5ee096cfba60/nixos/modules/hardware/logitech.nix#L75

Reply:
Thanks! I got it working and pushed the changes